src/iwant_bot/start.py

```diff
@@ -13,104 +14,134 @@
-    text += ['Or you can test POST request by command:']
-    text += ['> curl -X POST  --data "user_name=0&channel_id=1&channel_name=2']
-    text += ['  team_domain=4&team_id=5&text=6" http://localhost:8080']
-    return web.Response(text='\n'.join(text))
+async def handle_get(request):
+    """Handle GET request, can be display at http://localhost:8080"""
+    text = (f'Server is running at {request.url}.\n'
+            f'Try `curl -X POST --data "text=test" {request.url}example`\n')
+    return web.Response(text=text)
 
 
-def format_response(message='No response.'):
-    return json.dumps({'text': message})
+async def handle_other_posts(request):
+    """Handle all other POST requests.
+    For testing purpose, `curl -X POST --data "text=test" http://localhost:8080/example`"""
+    body = multidict_to_dict(await request.post())
+    print(f"INFO: The post to endpoint /{request.match_info['post']} contained:\n {body}")
+    return web.json_response({'text': f"POST to /{request.match_info['post']} is not resolved."})
 
 
-def verify_post_request(token):
-    """Slack `/commands` carry token, which must be same as VERIFICATION"""
-    return token == VERIFICATION
+def multidict_to_dict(multidict) -> dict:
+    if not len(set(multidict.keys())) == len(multidict):
+        print('WARNING: MultiDict contains duplicate keys, last occurrence was used.')
+        print(multidict)
+    return {key: multidict[key] for key in multidict}
 
 
-def body_to_dict(body):
-    """Try to convert MultiDictProxy to dictionary.
-    Not handled ValueError"""
-    keys = set(body.keys())
-    if len(keys) == len(body):
-        body_dict = {}
-        for key in keys:
-            body_dict[key] = body[key]
-    else:
-        raise ValueError('MultiDict contains same keys.')
-    return body_dict
+async def handle_slack_iwant(request):
+    body = multidict_to_dict(await request.post())
+    body['incoming_ts'] = time.time()
+    print(f'INFO: iwant request body:\n{body}.')
 
+    try:
+        verify_request_token(body)
+    except (KeyError, TokenError) as err:
+        print(f'INFO: Invalid token: {err}')
+        return web.json_response({'text': 'Unverified message.'})
 
-def trigger_reaction(body, trigger):
-    """Commands: /iwant,
-    Trigger_words: None"""
-    message = format_response(
-        f"{body['user_name']} used {body[trigger]} with {body['text']}.")
-    return web.json_response(body=message)
+    if 'command' in body:
+        print(f"INFO: iwant handler handles command '{body['command']}'")
+    else:
+        print("WARNING: Request does not specify 'command'.")
+        return web.json_response({'text': 'Tried to handle command, but none found.'})
+
+    iwant_object = IwantRequest(body, _iwant_activities, _iwant_behest, _slack_user_pattern,
+                                _other_words, _default_duration, _max_duration)
+    print(f'INFO: iwant parsed request:\n{iwant_object.data}')
```

Comment:
Worth noting, but out of scope for this PR: Those `print` calls can be substituted by calls to the `logging` module. (e.g. `logger.info('iwant parsed request: %s', iwant_object.data)`.